### ISSUE TITLE
Don't use txid as primary key part

### ISSUE THREAD

**Issue by elsirion** [elsirion/fedimint-observer#5]
We nearly never care about the txid, whereas the session index and item index in a session is a perfectly fine primary key for transactions too, is shorter and we even care about sesssion indices often (so it's practical if we don't have to join over transactions to find the session of a tx input/output).

https://github.com/elsirion/fedimint-observer/blob/de6aa05e16e029ccde37b2abf64b0b9d89fe04e6/schema/v0.sql#L35

**Comment by tacio** [comment]
Yeah, it happened to me once that the txid did not match between two different builds of fedimint_observer, when the federation_id, session_index, item_index (and everything else) were the same. So, I'd rather not use txid as part of keys
